ws-client/src/transport.rs

```diff
@@ -234,10 +235,15 @@ impl<'a> WsTransportClientBuilder<'a> {
 			&self.target.host_header,
 			&self.target.path_and_query,
 		);
+
+		let mut headers: ArrayVec<Header, 1> = ArrayVec::new();
+
 		if let Some(origin) = self.origin_header.as_ref() {
-			client.set_origin(origin);
+			headers.push(Header { name: "Origin", value: origin.as_bytes() });
 		}
 
+		client.set_headers(&headers);
```

Comment:
dq: why not do all this inside the `if let Some(origin)…` clause?

Reply:
`headers` have to be defined outside of the block to satisfy lifetimes, and it seemed more natural to me to set headers on the same scope they are defined.

Reply:
@maciejhirsz I did it slightly different in #397 but this is quite neat not re-export Header from this library but I did it with a single array. So not sure why ArrayVec dep is needed but maybe I did some refactoring with the redirect loop to make it work not sure.

```
let origin = self.origin_header.map(|o| [o]);
let client = some_code();
if let Some(origin) = self.origin_header.as_ref() {
   client.set_headers(origin);
}
...
```

Reply:
ah I see now, origin is borrowed and that won't work.